backend/capellacollab/core/exceptions.py

```diff
@@ -1,61 +1,53 @@
 # SPDX-FileCopyrightText: Copyright DB InfraGO AG and contributors
 # SPDX-License-Identifier: Apache-2.0
 
-import dataclasses
+import pydantic
+from fastapi import status
 
-import fastapi
-from fastapi import exception_handlers, status
 
-
-@dataclasses.dataclass
-class ExistingDependenciesError(Exception):
-    entity_name: str
-    entity_type: str
-    dependencies: list[str]
-
-
-async def existing_dependencies_exception_handler(
-    request: fastapi.Request, exc: ExistingDependenciesError
-) -> fastapi.Response:
-    dependencies_str = ", ".join(exc.dependencies)
-    return await exception_handlers.http_exception_handler(
-        request,
-        fastapi.HTTPException(
-            status_code=status.HTTP_409_CONFLICT,
-            detail={
-                "reason": [
-                    f"The {exc.entity_type} '{exc.entity_name}' can not be deleted. Please remove the following dependencies first: {dependencies_str}"
-                ]
-            },
-        ),
+@pydantic.dataclasses.dataclass
+class BaseError(Exception):
+    status_code: int = pydantic.Field(
+        description="The HTTP status code of an exception, accepts any int but passed as fastapi.status.DESIRED_STATUS_CODE for code readability",
+        examples=[status.HTTP_404_NOT_FOUND],
     )
-
-
-@dataclasses.dataclass
-class ResourceAlreadyExistsError(Exception):
-    resource_name: str
-    identifier_name: str
-
-
-async def resource_already_exists_exception_handler(
-    request: fastapi.Request, exc: ResourceAlreadyExistsError
-):
-    return await exception_handlers.http_exception_handler(
-        request,
-        fastapi.HTTPException(
-            status_code=status.HTTP_409_CONFLICT,
-            detail={
-                "reason": f"A {exc.resource_name} with a similar {exc.identifier_name} already exists.",
-                "technical": f"{exc.identifier_name} already used",
-            },
-        ),
+    title: str = pydantic.Field(
+        description="The title of the error, displayed in the frontend",
+        examples=["User not found"],
     )
-
-
-def register_exceptions(app: fastapi.FastAPI):
-    app.add_exception_handler(
-        ExistingDependenciesError, existing_dependencies_exception_handler  # type: ignore[arg-type]
+    reason: str = pydantic.Field(
+        description="The reason for the error and any possible resolutions/next steps, displayed in the frontend",
+        examples=["The user 'username' doesn't exist."],
     )
-    app.add_exception_handler(
-        ResourceAlreadyExistsError, resource_already_exists_exception_handler  # type: ignore[arg-type]
+    err_code: str | None = pydantic.Field(
```

Comment:
```suggestion
    err_code: str = pydantic.Field(
```

This PR is a good chance to make it required. Can you add error codes to all exceptions?